app/client-graphql-test/page.tsx

```diff
@@ -1,22 +1,32 @@
 "use client";
 
-import { useContributorRolesQuery } from '@/generated/graphql';
+import React, { useState, useEffect } from "react";
+import { useContributorRolesQuery } from "@/generated/graphql";
+import { handleApolloErrors } from "@/utils/gqlErrorHandler";
 
-/**
- * This is a test page to demo and test the use of graphql hooks on the client side.
- * Client-side graphql requests uses the apollo-wrapper.tsx file
- * @returns 
- */
 export default function Page() {
+    const [errors, setErrors] = useState<string[]>([]);
+    const { data, loading, error, refetch } = useContributorRolesQuery();
     let roles;
-    const { data, loading, error } = useContributorRolesQuery();
-    if (loading) {
-        return <div>Loading...</div>
-    }
 
-    if (error) {
-        console.error(error);
-        return <div>Error</div>
+    // UseEffect to handle async error handling
+    useEffect(() => {
+        if (error) {
+            const handleErrors = async () => {
```

Comment:
Update this graphql test page to call a shared handleApolloErrors function. I moved some of the error handling out of errorLink and into this function so we could have better control over refetching the data for the page after refreshing auth tokens.